src/App.js

```diff
@@ -7,23 +8,30 @@ import Deals from "./pages/Deals";
 import Item from "./pages/Item";
 
 export default function App() {
+  useEffect(() => {
```

Comment:
It would be better to make this query on the individual pages since they all request data a little bit differently.